contribs/gnodev/pkg/events/middleware.go

```diff
@@ -0,0 +1,105 @@
+package events
+
+import (
+	"bytes"
+	_ "embed"
+	"net/http"
+	"strings"
+	"sync"
+	"text/template"
+)
+
+//go:embed static/hotreload.js
+var reloadscript string
+
+type Middleware struct {
+	remote   string
+	muRemote sync.RWMutex
+
+	next     http.Handler
+	tmpl     *template.Template
+	onceExec *sync.Once
+	script   []byte
+}
+
+func NewMiddleware(remote string, next http.Handler) *Middleware {
```

Comment:
Don't export `Middleware`, add godoc, make this return a `http.Handler` instead.